src/Jobs/Recruit.Vacancies.Jobs/Triggers/QueueTriggers/DeleteStaleQueryStoreDocumentsQueueTrigger.cs

```diff
@@ -0,0 +1,87 @@
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.EditVacancyInfo;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.Employer;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.Provider;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.Vacancy;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.StorageQueue;
+using Esfa.Recruit.Vacancies.Jobs.Configuration;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Esfa.Recruit.Vacancies.Jobs.Triggers.QueueTriggers
+{
+    public class DeleteStaleQueryStoreDocumentsQueueTrigger
+    {
+        private readonly ILogger<DeleteStaleQueryStoreDocumentsQueueTrigger> _logger;
+        private readonly RecruitWebJobsSystemConfiguration _jobsConfig;
+        private readonly ITimeProvider _timeProvider;
+        private readonly IQueryStoreHouseKeepingService _queryStoreHouseKeepingService;
+        private string JobName => GetType().Name;
+
+        private const int DefaultStaleByDays = 90;
+
+        public DeleteStaleQueryStoreDocumentsQueueTrigger(ILogger<DeleteStaleQueryStoreDocumentsQueueTrigger> logger, 
+            RecruitWebJobsSystemConfiguration jobsConfig,
+            ITimeProvider timeProvider,
+            IQueryStoreHouseKeepingService queryStoreHouseKeepingService)
+        {
+            _logger = logger;
+            _jobsConfig = jobsConfig;
+            _timeProvider = timeProvider;
+            _queryStoreHouseKeepingService = queryStoreHouseKeepingService;
+        }
+
+        public async Task DeleteStaleQueryStoreDocumentsAsync([QueueTrigger(QueueNames.DeleteStaleQueryStoreDocumentsQueueName, Connection = "QueueStorage")] string message, TextWriter log)
+        {
+            try
+            {
+                if (_jobsConfig.DisabledJobs.Contains(JobName))
+                {
+                    _logger.LogDebug($"{JobName} is disabled, skipping ...");
+                    return;
+                }
+
+                var payload = JsonConvert.DeserializeObject<DeleteStaleQueryStoreDocumentsQueueMessage>(message);
+
+                var targetDate = payload?.CreatedByScheduleDate ?? _timeProvider.Today;
+
+                var documentsStaleByDate = targetDate.AddDays((_jobsConfig.QueryStoreDocumentsStaleByDays ?? DefaultStaleByDays) * -1);
+                
+                _logger.LogInformation($"Begining to delete query store stale documents that have not been updated since {documentsStaleByDate.ToShortDateString()}");
+
+                var documentTypesToDelete = new [] 
+                {
+                    nameof(EditVacancyInfo), 
+                    nameof(ProviderEditVacancyInfo),
+                    nameof(EmployerDashboard),
+                    nameof(ProviderDashboard),
+                    nameof(ClosedVacancy)
+                };
+
+                foreach (var viewType in documentTypesToDelete)
```

Comment:
naming? 

`documentType in documentTypesToDelete`

Reply:
This change is not part of this implementation. I just realised this was wrongly based on `master` branch. I have fixed the base. Also I will fix the naming in the respective branch.